### PR TITLE
unify lock controller / logic / rendering

### PR DESCRIPTION
**Pull request by grosser**
@pswadi-zendesk @sandlerr @jonmoter @jwswj 

 - allow description for global locks
 - unify lock forms
 - unify lock display

### CODE REVIEW

File: app/controllers/locks_controller.rb [hunk 1]
```diff
@@ -1,31 +1,32 @@
...
+    redirect_to :back, notice: "Unlocked"
   end
 
   protected
 
-  def project
-    @project ||= Project.find_by_param!(params[:project_id])
+  def for_global_lock?
+    case action_name
+    when "create" then !params[:lock].try(:[], :stage_id)
```
Comment (jwswj) [inline app/controllers/locks_controller.rb]: What does `:[]` do?

**Comment by jwswj** [comment]
Like it @grosser :+1:

File: app/controllers/locks_controller.rb (hunk 1 above)
Comment (grosser) [inline app/controllers/locks_controller.rb]: x[1] -> x.send(:[], 1)

**Comment by grosser** [comment]
@zendesk/runway test

**Pull request merged**